src/backgroundblurvideoframeprocessor/BackgroundBlurVideoFrameProcessor.ts

```diff
@@ -0,0 +1,257 @@
+    const u = new URL(url);
+    const keys = Object.keys(queryParams);
+    for (const key of keys) {
+      if (queryParams[key] !== undefined) {
+        u.searchParams.set(key, queryParams[key]);
+      }
+    }
+
+    return u.toString();
+  }
+
+  /**
+   * Based on the spec that is passed in set defaults for spec
+   * @param spec the spec that was passed in
+   * @returns An updated spec with defaults set
+   */
+  private static resolveSpec(spec?: BackgroundFilterSpec): BackgroundFilterSpec {
+    const {
+      paths = DEFAULT_PATHS,
+      model = CREATE_DEFAULT_MODEL_SPEC(),
+      assetGroup = this.defaultAssetSpec().assetGroup,
+      revisionID = this.defaultAssetSpec().revisionID,
+    } = spec || {};
+
+    const params = {
+      assetGroup,
+      revisionID,
+      sdk: encodeURIComponent(Versioning.sdkVersion),
+      ua: encodeURIComponent(Versioning.sdkUserAgentLowResolution),
+    };
+
+    paths.worker = this.createUrlWithParams(paths.worker, params);
+    paths.wasm = this.createUrlWithParams(paths.wasm, params);
+    paths.simd = this.createUrlWithParams(paths.simd, params);
+    model.path = this.createUrlWithParams(model.path, params);
+
+    return {
+      paths,
+      model,
+      assetGroup,
+      revisionID,
+    };
+  }
+
+  /**
+   * Based on the options that are passed in set defaults for options
+   * @param options  the options that are passed in
+   * @returns An updated set of options with defaults set
+   */
+  private static resolveOptions(options?: BackgroundBlurOptions): BackgroundBlurOptions {
+    options = options ?? {};
+    if (!options.blurStrength) {
+      options.blurStrength = BlurStrength.MEDIUM;
+    }
+
+    if (!options.logger) {
+      options.logger = new ConsoleLogger('BackgroundBlurProcessor', LogLevel.INFO);
+    }
+
+    if (!options.reportingPeriodMillis) {
+      options.reportingPeriodMillis = 1000;
+    }
+    return options;
+  }
+
+  /**
+   * This method will detect the environment in which it is being used and determine if background
+   * blur can be used.
+   * @param spec The {@link BackgroundBlurSpec} spec that will be used to initialize asssets
+   * @param options options such as logger
+   * @returns a boolean promise that will resolve to true if supported and false if not
+   */
+  static isSupported(spec?: BackgroundFilterSpec, options?: { logger?: Logger }): Promise<boolean> {
+    spec = BackgroundBlurVideoFrameProcessor.resolveSpec(spec);
+    options = BackgroundBlurVideoFrameProcessor.resolveOptions(options);
+    const { logger } = options;
+
+    // could not figure out how to remove globalThis to test failure case
+    /* istanbul ignore next */
+    if (typeof globalThis === 'undefined') {
+      logger.info('Browser does not have globalThis.');
+      return Promise.resolve(false);
+    }
+
+    const browser = new DefaultBrowserBehavior();
+    if (browser.name() === 'safari' && browser.majorVersion() === 15) {
```

Comment:
we may want to have a browser behavior function here instead. Could you please leave a comment to the webkit bug if there is one so that we know why we have this check and when we can remove it in the future.